src/array.cr

```diff
@@ -1770,4 +1770,12 @@ class Array(T)
       end
     end
   end
+
+  def bsearch
+    idx=(0...self.size).bsearch{|i|yield self[i]}
```

Comment:
[Fixed] I was able to pass the block `bsearch(&block : T -> Int|Float|Bool?)` to bsearch_index...